consensus/clique/src/main/java/tech/pegasys/pantheon/consensus/clique/CliqueBlockHashing.java

```diff
@@ -48,6 +49,9 @@ public static Hash calculateDataHashForProposerSeal(
    */
   public static Address recoverProposerAddress(
       final BlockHeader header, final CliqueExtraData cliqueExtraData) {
+    if (header.getNumber() == 0) {
```

Comment:
```suggestion
    if (header.getNumber() == BlockHeader.GENESIS_BLOCK_NUMBER) {
```

Please don't ever hard code 0 as the genesis block number. One day we're going to have to support variable genesis block numbers.

Reply:
done